### DISCUSSION TITLE
Creating a HTML component

### DISCUSSION THREAD

**Discussion by cmeffley** [nss-evening-cohort-14/evening-client-side#106] [Help Me!] [answered]
Chris Meffley

No other help tickets

I've been trying to place the sorting hat html in an JS file and create a function to get it to render to the DOM, but without success.
The things I've found seem to be a lot more complicated than what I think is needed.
Error message: Cannot call a class as a function

https://css-tricks.com/crafting-reusable-html-templates/
https://www.youtube.com/watch?v=PCWaFLy3VUo
https://developer.mozilla.org/en-US/docs/Web/API/Document/createElement

I've tried creating classes, templates, things with shadow DOM (don't completely understand that, so trying to stay away from it). Also, created a regular form and called the renderToDOM function. I've spent way too much time and definitely need some guidance.

file links:
loadScreen to html: https://github.com/cmeffley/sorting-hat-webpack/blob/main/src/javascripts/components/loadScreen.js  (got a new error of cannot set property 'innerhtml' of null)
all JS: https://github.com/cmeffley/sorting-hat-webpack/tree/main/src/javascripts
index.html: https://github.com/cmeffley/sorting-hat-webpack/blob/main/src/index.html

**Comment by Gabrielle-Tobermann** [comment]
I don't know if this is the right way to do it but what I did was, I put the html stuff inside a variable and then called renderToDOM with the body and the variable as parameters.

**Comment by drteresavasquez** [comment]
Where is the link to your code? Make sure that you follow the template for submitting so that we have all the info. If we don't have it, we won't answer the ticket.

**Reply by cmeffley** [reply]
Added links, my apologies

**Comment by TrinityChristiana** [comment]
@cmeffley Can you add your solution to the issue?

**Comment by cmeffley** [comment, marked as answer]
Got an error of "TypeError cannot set property "innerHTML" of null coming from the renderToDom.js. Did a console.warn of  selectedDiv and got null, so did a console.warn of divId and got undefined.  But there was also an error message coming from main.js. Looked at where the domEvents function was being called, and upon closer look, realized that I was trying to import the domEvents from the wrong module. So, corrected the path to './helpers/data/domEvents.'
I learned that I need to become better friends with using console to help direct me to hopefully the answer, not just google.
